tests/TestCli.cpp

```diff
@@ -250,106 +259,80 @@ void TestCli::testAdd()
     QVERIFY(!addCmd.name.isEmpty());
     QVERIFY(addCmd.getDescriptionLine().contains(addCmd.name));
 
-    Utils::Test::setNextPassword("a");
-    addCmd.execute({"add",
-                    "-u",
-                    "newuser",
-                    "--url",
-                    "https://example.com/",
-                    "-g",
-                    "-L",
-                    "20",
-                    m_dbFile->fileName(),
-                    "/newuser-entry"});
-    m_stderrFile->reset();
-    m_stdoutFile->reset();
-    m_stdoutFile->readLine(); // skip password prompt
-    QCOMPARE(m_stderrFile->readAll(), QByteArray(""));
-    QCOMPARE(m_stdoutFile->readAll(), QByteArray("Successfully added entry newuser-entry.\n"));
-
-    auto db = readTestDatabase();
+    setInput("a");
+    execCmd(addCmd,
+            {"add",
+             "-u",
+             "newuser",
+             "--url",
+             "https://example.com/",
+             "-g",
+             "-L",
+             "20",
+             m_dbFile->fileName(),
+             "/newuser-entry"});
+    QVERIFY(m_stderr->readAll().contains("Enter password to unlock"));
+    QVERIFY(m_stdout->readAll().contains("Successfully added entry newuser-entry."));
+
+    auto db = readDatabase();
     auto* entry = db->rootGroup()->findEntryByPath("/newuser-entry");
     QVERIFY(entry);
     QCOMPARE(entry->username(), QString("newuser"));
     QCOMPARE(entry->url(), QString("https://example.com/"));
     QCOMPARE(entry->password().size(), 20);
 
     // Quiet option
-    qint64 pos = m_stdoutFile->pos();
-    qint64 posErr = m_stderrFile->pos();
-    Utils::Test::setNextPassword("a");
-    addCmd.execute({"add", "-q", "-u", "newuser", "-g", "-L", "20", m_dbFile->fileName(), "/newentry-quiet"});
-    m_stdoutFile->seek(pos);
-    m_stderrFile->seek(posErr);
-    QCOMPARE(m_stderrFile->readAll(), QByteArray(""));
-    QCOMPARE(m_stdoutFile->readAll(), QByteArray(""));
-    db = readTestDatabase();
+    setInput("a");
+    execCmd(addCmd, {"add", "-q", "-u", "newuser", "-g", "-L", "20", m_dbFile->fileName(), "/newentry-quiet"});
+    QCOMPARE(m_stdout->readAll(), QByteArray(""));
```

Comment:
@droidmonkey why not preserve the `m_stderr` assertion?

Reply:
m_stderr now contains the password prompt message so you can't test for blank.

Reply:
We used to skip the password prompt with a `readLine()` and then assert that the next line was empty. This would assert that nothing else was written to `stderr` other than the password prompt.